### ISSUE TITLE
Pool not detected

### ISSUE THREAD

**Issue by mrnickgbk** [nkshirsagar/thinpool_shrink#1]
Hello, 

Using ./thin_shrink.py -L 460G -t data-volume-group/data-thin-pool

I get this error:

Warning: No tpool device found, perhaps pool has no thins?

But these are the correct vg and pool name:

From running lvs
LV                             VG                 Attr       LSize   Pool           Origin         Data%  Meta%  
data-thin-pool                 data-volume-group  twi-aotz-- 465.67g                            

Could it be the dashes causing an issue with the name? dmstatus reports the name with --

**Comment by nkshirsagar** [comment]
Are there any thin logical volumes created in the pool ?

And this tool is not yet integrated into lvm2 so the freed up space will
not be released back to the volume group today.

Secondly, if there are snapshots in the thin pool, it's buggy.

So be aware this is wip BUT I appreciate that you're testing it out ..

**Comment by mrnickgbk** [comment]
Appreciate the quick reply.

There are thin lv's yes

I'm using RHEL7 (LVM2). So you say the free'd up space will not be released? I was looking to reclaim some extents to increase metadata partition. Is that nor possible with this tool then?

And yes there are snapshots.

Thanks!

**Comment by nkshirsagar** [comment]
No, this tool will be rewritten in rust and then integrated with lvm2 so
that the tdata reduction is then possible (this tool does not resize tdata
today gracefully.. it can do it with a forced vgcfgrestore with a changed
mapping in the vg metadata. But for the released space to be given back to
the volume group needs integration in lvm2. It's planned for a future
release.

I'm not sure though why it's not working on your setup if it has thin
logical volumes. I'll debug it later if you send me lvs -a and a console
output..

**Comment by mrnickgbk** [comment]
Thanks for the info .

 Ah so it would not help me anyhow, as I need to reclaim extents. I'd be interested in the vgconfigrestore option if that was available / documented - this is a production system and your tool is pretty much my last resort before having to blow away the lvm config and data :(

I'll happily send you the lvs output but I'd prefer not to post it here for all to see. Is there an email I can contact you on?

**Comment by nkshirsagar** [comment]
Nkshirsa@redhat.com

And yes we can discuss further over email and I can put you in touch with
the upstream maintainer who's working on the rust verson and you could get
more information about the eventual release upstream with lvm integration
(tdata reduction and vg space reclaim).
This tool currently does metadata changes and data copying to make the pool
shrinkable. The rest is just a demo to show the reduced size currently with
a forced vgcfgrestore, not meant to be the final approach for the space to
be seen in the vg.  lvreduce will do the lvm2 and dm side changes for
proper space reclaim to the volume group. This is all wip.
